packages/amplify_auth_plugin_interface/lib/src/Password/ResetPasswordRequest.dart

```diff
@@ -27,7 +27,7 @@ class ResetPasswordRequest {
       pendingRequest['username'] = username;
     }
     if (options != null) {
-      pendingRequest['options'] = options.serializeAsMap();
+      pendingRequest['options'] = options!.serializeAsMap();
     }
```

Comment:
Not necessary I guess? You would have gotten a warning from dart analyzer https://dart.dev/null-safety/understanding-null-safety#unnecessary-code-warnings

You can run the dart analysis and remove similar code.

Reply:
Hmm I do not get warnings from the dart analyzer. 

Why would this be unnecessary? 

As options is a nullable type we need to make sure it's not null before getting its value.  If we just called options!.serializeAsMap() if options was null the program would crash.

Reply:
I'm talking about the `!` being unnecessary since we already established  on line 29 that options is non-nullable so force unwrapping shouldn't be required

Reply:
Unfortunately the compiler isn't smart enough.  Removing the ! will result in an error as it thinks options can still be null.